domain/machine/state/reboot.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	machineerrors "github.com/juju/juju/domain/machine/errors"
+	"github.com/juju/juju/internal/database"
+)
+
+// RebootAction defines the action a machine should
+// take when a hook needs to reboot
+type RebootAction string
+
+const (
+	// ShouldDoNothing instructs a machine agent that no action
+	// is required on its part
+	ShouldDoNothing RebootAction = "noop"
+	// ShouldReboot instructs a machine to reboot
+	// this happens when a hook running on a machine, requests
+	// a reboot
+	ShouldReboot RebootAction = "reboot"
+	// ShouldShutdown instructs a machine to shut down. This usually
+	// happens when running inside a container, and a hook on the parent
+	// machine requests a reboot
+	ShouldShutdown RebootAction = "shutdown"
+)
```

Comment:
Move this to `domain/machine/types.go` then we can share this between service and state. Service shouldn't know about state and state shouldn't know about service.